sklearn/model_selection/_classification_threshold.py

```diff
@@ -0,0 +1,924 @@
+        super().__init__(
+            score_func=score_func,
+            sign=sign,
+            kwargs=kwargs,
+            response_method=response_method,
+        )
+        self._n_thresholds = n_thresholds
+
+    @classmethod
+    def from_scorer(cls, scorer, response_method, n_thresholds, pos_label):
+        """Create a continuous scorer from a normal scorer."""
+        # add `pos_label` if requested by the scorer function
+        scorer_kwargs = {**scorer._kwargs}
+        signature_scoring_func = signature(scorer._score_func)
+        if (
+            "pos_label" in signature_scoring_func.parameters
+            and "pos_label" not in scorer_kwargs
+        ):
+            if pos_label is None:
+                # Since the provided `pos_label` is the default, we need to
+                # use the default value of the scoring function that can be either
+                # `None` or `1`.
+                scorer_kwargs["pos_label"] = signature_scoring_func.parameters[
+                    "pos_label"
+                ].default
+            else:
+                scorer_kwargs["pos_label"] = pos_label
+        # transform a binary metric into a curve metric for all possible decision
+        # thresholds
+        instance = cls(
+            score_func=scorer._score_func,
+            sign=scorer._sign,
+            response_method=response_method,
+            n_thresholds=n_thresholds,
+            kwargs=scorer_kwargs,
+        )
+        # transfer the metadata request
+        instance._metadata_request = scorer._get_metadata_request()
+        return instance
+
+    def _score(self, method_caller, estimator, X, y_true, **kwargs):
+        """Evaluate predicted target values for X relative to y_true.
+
+        Parameters
+        ----------
+        estimator : object
+            Trained estimator to use for scoring.
+
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Test data that will be fed to estimator.predict.
+
+        y_true : array-like of shape (n_samples,)
+            Gold standard target values for X.
+
+        **kwargs : dict
+            Other parameters passed to the scorer. Refer to
+            :func:`set_score_request` for more details.
+
+        Returns
+        -------
+        scores : ndarray of shape (n_thresholds,)
+            The scores associated to each threshold.
+
+        potential_thresholds : ndarray of shape (n_thresholds,)
+            The potential thresholds used to compute the scores.
+        """
+        pos_label = self._get_pos_label()
+        y_score = method_caller(
+            estimator, self._response_method, X, pos_label=pos_label
+        )
+
+        scoring_kwargs = {**self._kwargs, **kwargs}
+        if isinstance(self._n_thresholds, Integral):
+            potential_thresholds = np.linspace(
+                np.min(y_score), np.max(y_score), self._n_thresholds
+            )
```

Comment:
When `y_score` is probability then this would be possible. If this is the output of `decision_function` then this is more complicated because we don't have finite bounds. So I'm not sure how to define them before to call `method_caller` within the function.

Do you think of an alternative where we could define the extremum if `method_caller` is `decision_function`.

Reply:
Hum, I was only thinking about `predict_proba` :(

> So I'm not sure how to define them before to call method_caller within the function.

It's possible using a bijection from [-inf, inf] to [0, 1] but it won't make things simpler, just move the complexity, so let's forget that.